### ISSUE TITLE
clean_style_tag() generating "Undefined offset" notice when CleanUp is enabled

### ISSUE THREAD

**Issue by jimmyandrade** [roots/soil#100]
I'm getting the following error.

PS.: I'm using only one widget (JetPack Facebook like box) and clean-up is enabled.

<font size='1'><table class='xdebug-error xe-notice' dir='ltr' border='1' cellspacing='0' cellpadding='1'>
<tr><th align='left' bgcolor='#f57900' colspan="5"><span style='background-color: #cc0000; color: #fce94f; font-size: x-large;'>( ! )</span> Notice: Undefined offset: 0 in app\plugins\soil\modules\clean-up.php on line <i>103</i></th></tr>
<tr><th align='left' bgcolor='#e9b96e' colspan='5'>Call Stack</th></tr>
<tr><th align='center' bgcolor='#eeeeec'>#</th><th align='left' bgcolor='#eeeeec'>Time</th><th align='left' bgcolor='#eeeeec'>Memory</th><th align='left' bgcolor='#eeeeec'>Function</th><th align='left' bgcolor='#eeeeec'>Location</th></tr>
<tr><td bgcolor='#eeeeec' align='center'>1</td><td bgcolor='#eeeeec' align='center'>0.0004</td><td bgcolor='#eeeeec' align='right'>240088</td><td bgcolor='#eeeeec'>{main}(  )</td><td title='index.php' bgcolor='#eeeeec'>..\index.php<b>:</b>0</td></tr>
<tr><td bgcolor='#eeeeec' align='center'>2</td><td bgcolor='#eeeeec' align='center'>0.0009</td><td bgcolor='#eeeeec' align='right'>243232</td><td bgcolor='#eeeeec'>require( <font color='#00bb00'>'wp\wp-blog-header.php'</font> )</td><td title='index.php' bgcolor='#eeeeec'>..\index.php<b>:</b>5</td></tr>
<tr><td bgcolor='#eeeeec' align='center'>3</td><td bgcolor='#eeeeec' align='center'>0.3542</td><td bgcolor='#eeeeec' align='right'>28930776</td><td bgcolor='#eeeeec'>require_once( <font color='#00bb00'>'wp\wp-includes\template-loader.php'</font> )</td><td title='wp\wp-blog-header.php' bgcolor='#eeeeec'>..\wp-blog-header.php<b>:</b>16</td></tr>
<tr><td bgcolor='#eeeeec' align='center'>4</td><td bgcolor='#eeeeec' align='center'>0.3574</td><td bgcolor='#eeeeec' align='right'>29264280</td><td bgcolor='#eeeeec'>include( <font color='#00bb00'>'app\themes\dreamville\index.php'</font> )</td><td title='wp\wp-includes\template-loader.php' bgcolor='#eeeeec'>..\template-loader.php<b>:</b>74</td></tr>
<tr><td bgcolor='#eeeeec' align='center'>5</td><td bgcolor='#eeeeec' align='center'>0.3951</td><td bgcolor='#eeeeec' align='right'>30174904</td><td bgcolor='#eeeeec'>get_footer(  )</td><td title='app\themes\dreamville\index.php' bgcolor='#eeeeec'>..\index.php<b>:</b>150</td></tr>
<tr><td bgcolor='#eeeeec' align='center'>6</td><td bgcolor='#eeeeec' align='center'>0.3951</td><td bgcolor='#eeeeec' align='right'>30175544</td><td bgcolor='#eeeeec'>locate_template(  )</td><td title='wp\wp-includes\general-template.php' bgcolor='#eeeeec'>..\general-template.php<b>:</b>85</td></tr>
<tr><td bgcolor='#eeeeec' align='center'>7</td><td bgcolor='#eeeeec' align='center'>0.3952</td><td bgcolor='#eeeeec' align='right'>30175720</td><td bgcolor='#eeeeec'>load_template(  )</td><td title='wp\wp-includes\template.php' bgcolor='#eeeeec'>..\template.php<b>:</b>477</td></tr>
<tr><td bgcolor='#eeeeec' align='center'>8</td><td bgcolor='#eeeeec' align='center'>0.3954</td><td bgcolor='#eeeeec' align='right'>30189144</td><td bgcolor='#eeeeec'>require_once( <font color='#00bb00'>'app\themes\dreamville\footer.php'</font> )</td><td title='wp\wp-includes\template.php' bgcolor='#eeeeec'>..\template.php<b>:</b>501</td></tr>
<tr><td bgcolor='#eeeeec' align='center'>9</td><td bgcolor='#eeeeec' align='center'>0.3954</td><td bgcolor='#eeeeec' align='right'>30189176</td><td bgcolor='#eeeeec'>wp_footer(  )</td><td title='app\themes\dreamville\footer.php' bgcolor='#eeeeec'>..\footer.php<b>:</b>11</td></tr>
<tr><td bgcolor='#eeeeec' align='center'>10</td><td bgcolor='#eeeeec' align='center'>0.3954</td><td bgcolor='#eeeeec' align='right'>30189256</td><td bgcolor='#eeeeec'>do_action(  )</td><td title='wp\wp-includes\general-template.php' bgcolor='#eeeeec'>..\general-template.php<b>:</b>2220</td></tr>
<tr><td bgcolor='#eeeeec' align='center'>11</td><td bgcolor='#eeeeec' align='center'>0.3954</td><td bgcolor='#eeeeec' align='right'>30190736</td><td bgcolor='#eeeeec'><a href='http://www.php.net/function.call-user-func-array:{wp\wp-includes\plugin.php:496}' target='_new'>call_user_func_array:{wp\wp-includes\plugin.php:496}</a>
(  )</td><td title='wp\wp-includes\plugin.php' bgcolor='#eeeeec'>..\plugin.php<b>:</b>496</td></tr>
<tr><td bgcolor='#eeeeec' align='center'>12</td><td bgcolor='#eeeeec' align='center'>0.3954</td><td bgcolor='#eeeeec' align='right'>30190800</td><td bgcolor='#eeeeec'>wp_print_footer_scripts(  )</td><td title='wp\wp-includes\plugin.php' bgcolor='#eeeeec'>..\plugin.php<b>:</b>496</td></tr>
<tr><td bgcolor='#eeeeec' align='center'>13</td><td bgcolor='#eeeeec' align='center'>0.3955</td><td bgcolor='#eeeeec' align='right'>30190888</td><td bgcolor='#eeeeec'>do_action(  )</td><td title='wp\wp-includes\script-loader.php' bgcolor='#eeeeec'>..\script-loader.php<b>:</b>953</td></tr>
<tr><td bgcolor='#eeeeec' align='center'>14</td><td bgcolor='#eeeeec' align='center'>0.3955</td><td bgcolor='#eeeeec' align='right'>30192384</td><td bgcolor='#eeeeec'><a href='http://www.php.net/function.call-user-func-array:{wp\wp-includes\plugin.php:496}' target='_new'>call_user_func_array:{wp\wp-includes\plugin.php:496}</a>
(  )</td><td title='wp\wp-includes\plugin.php' bgcolor='#eeeeec'>..\plugin.php<b>:</b>496</td></tr>
<tr><td bgcolor='#eeeeec' align='center'>15</td><td bgcolor='#eeeeec' align='center'>0.3955</td><td bgcolor='#eeeeec' align='right'>30192448</td><td bgcolor='#eeeeec'>_wp_footer_scripts(  )</td><td title='wp\wp-includes\plugin.php' bgcolor='#eeeeec'>..\plugin.php<b>:</b>496</td></tr>
<tr><td bgcolor='#eeeeec' align='center'>16</td><td bgcolor='#eeeeec' align='center'>0.3955</td><td bgcolor='#eeeeec' align='right'>30192552</td><td bgcolor='#eeeeec'>print_late_styles(  )</td><td title='wp\wp-includes\script-loader.php' bgcolor='#eeeeec'>..\script-loader.php<b>:</b>938</td></tr>
<tr><td bgcolor='#eeeeec' align='center'>17</td><td bgcolor='#eeeeec' align='center'>0.3955</td><td bgcolor='#eeeeec' align='right'>30192800</td><td bgcolor='#eeeeec'>WP_Styles->do_footer_items(  )</td><td title='wp\wp-includes\script-loader.php' bgcolor='#eeeeec'>..\script-loader.php<b>:</b>1017</td></tr>
<tr><td bgcolor='#eeeeec' align='center'>18</td><td bgcolor='#eeeeec' align='center'>0.3955</td><td bgcolor='#eeeeec' align='right'>30192896</td><td bgcolor='#eeeeec'>WP_Dependencies->do_items(  )</td><td title='wp\wp-includes\class.wp-styles.php' bgcolor='#eeeeec'>..\class.wp-styles.php<b>:</b>238</td></tr>
<tr><td bgcolor='#eeeeec' align='center'>19</td><td bgcolor='#eeeeec' align='center'>0.3956</td><td bgcolor='#eeeeec' align='right'>30193280</td><td bgcolor='#eeeeec'>WP_Styles->do_item(  )</td><td title='wp\wp-includes\class.wp-dependencies.php' bgcolor='#eeeeec'>..\class.wp-dependencies.php<b>:</b>122</td></tr>
<tr><td bgcolor='#eeeeec' align='center'>20</td><td bgcolor='#eeeeec' align='center'>0.3961</td><td bgcolor='#eeeeec' align='right'>30194024</td><td bgcolor='#eeeeec'>apply_filters(  )</td><td title='wp\wp-includes\class.wp-styles.php' bgcolor='#eeeeec'>..\class.wp-styles.php<b>:</b>91</td></tr>
<tr><td bgcolor='#eeeeec' align='center'>21</td><td bgcolor='#eeeeec' align='center'>0.3981</td><td bgcolor='#eeeeec' align='right'>30196216</td><td bgcolor='#eeeeec'><a href='http://www.php.net/function.call-user-func-array:{wp\wp-includes\plugin.php:213}' target='_new'>call_user_func_array:{wp\wp-includes\plugin.php:213}</a>
(  )</td><td title='wp\wp-includes\plugin.php' bgcolor='#eeeeec'>..\plugin.php<b>:</b>213</td></tr>
<tr><td bgcolor='#eeeeec' align='center'>22</td><td bgcolor='#eeeeec' align='center'>0.3981</td><td bgcolor='#eeeeec' align='right'>30196248</td><td bgcolor='#eeeeec'>Roots\Soil\CleanUp\clean_style_tag(  )</td><td title='wp\wp-includes\plugin.php' bgcolor='#eeeeec'>..\plugin.php<b>:</b>213</td></tr>
</table></font>
<br />

**Comment by Foxaii** [comment]
Can you change line 103 in your clean-up.php to the following and report back please:

```php
$media = isset($matches[3][0]) && $matches[3][0] !== 'all' ? ' media="' . $matches[3][0] . '"' : '';
```

**Comment by Foxaii** [comment]
Thanks for the additional debugging and offer of a PR. It's really appreciated.

**Comment by retlehs** [comment]
closed by #102
